magma_access_gateway_installer/agw_installer_ubuntu.py

```diff
@@ -0,0 +1,95 @@
+
+from .agw_installation_errors import InvalidNumberOfInterfacesError, UnsupportedOSError
+
+logger = logging.getLogger(__name__)
+logger.addHandler(JournalHandler())
+logger.setLevel(logging.INFO)
+
+
+class AGWInstallerUbuntu:
+    """Main class of Magma Access Gateway installer for Ubuntu."""
+
+    MAGMA_USER = "magma"
+    REQUIRED_NUMBER_OF_NICS = 2
+
+    def __init__(self):
+        self.network_interfaces = netifaces.interfaces()
+        self.network_interfaces.remove("lo")
+
+    def _preinstall_checks(self):
+        """Checks whether installation preconditions are met. If not, relevant errors are being
+        logged and installation is cancelled.
+
+        :raises:
+            UnsupportedOSException: if OS is not Ubuntu
+            InvalidNumberOfInterfaces: if number of available network interfaces is different from
+                expected
+        """
+        if not self._check_if_ubuntu_is_installed():
+            logger.error("Invalid OS! \n Magma AGW can only be installed on Ubuntu! Exiting...")
+            raise UnsupportedOSError()
+        elif not self._check_if_required_amount_of_network_interfaces_is_available():
+            logger.error(
+                "Invalid number of network interfaces!"
+                "Magma AGW needs two network interfaces - SGi and S1! Exiting..."
+            )
+            raise InvalidNumberOfInterfacesError()
+        else:
+            logger.info("Magma AGW pre-install checks completed. Starting installation...")
+
+    def _magma_service_user_creation(self):
+        """Creates and configures Magma service user if necessary."""
+        if not self._check_if_magma_user_exists():
+            self._create_magma_user()
+            self._configure_magma_user()
+
+    @staticmethod
+    def _check_if_ubuntu_is_installed():
+        """Checks whether installed OS is Ubuntu."""
+        return "Ubuntu" in platform.version()
+
+    def _check_if_required_amount_of_network_interfaces_is_available(self):
+        """Checks whether required amount of network interfaces has been attached."""
+        return len(self.network_interfaces) == self.REQUIRED_NUMBER_OF_NICS
+
+    def _check_if_magma_user_exists(self):
+        """Checks whether Magma user exists."""
+        try:
+            pwd.getpwnam(self.MAGMA_USER)
+            return True
+        except KeyError:
+            logger.warning("Magma user not found!")
+            return False
+
+    def _create_magma_user(self):
+        """Adds Magma user."""
+        logger.info(f"Adding Magma user: {self.MAGMA_USER}...")
+        os.system(f'adduser --disabled-password --gecos "" {self.MAGMA_USER}')
+
+    def _configure_magma_user(self):
+        """Adds Magma user to sudoers."""
+        logger.info(f'Adding Magma user "{self.MAGMA_USER}" to sudoers...')
+        os.system(f"adduser ${self.MAGMA_USER} sudo")
+        with open("/etc/sudoers", "w") as sudoers:
+            sudoers.write(f"{self.MAGMA_USER} ALL=(ALL) NOPASSWD:ALL")
```

Comment:
Is this needed?

Reply:
Don't know yet. For now, I'm trying to follow **agw_install_ubuntu.sh**. 
I put this on my check list and will get back to it once whole script is rewritten to Python. If it turns out irrelevant, I'll remove it. 
Would that work for you?

Reply:
Yupp